components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java

```diff
@@ -192,6 +195,133 @@ public AuthenticationContext authenticate(String jwtToken, MessageContext synCtx
                 "Invalid JWT token. Signature verification failed.");
     }
 
+    /**
+     * Authenticates the given WebSocket handshake request with a JWT token to see if an API consumer is allowed to
+     * access a particular API or not.
+     *
+     * @param jwtToken   The JWT token sent with the API request
+     * @param apiContext The context of the invoked API
+     * @param apiVersion The version of the invoked API
+     * @return an AuthenticationContext object which contains the authentication information
+     * @throws APISecurityException in case of authentication failure
+     */
+    @MethodStats
+    public AuthenticationContext authenticateForWebSocket(String jwtToken, String apiContext, String apiVersion)
```

Comment:
We might have to add the revocation logic as well in here

Reply:
We can add it once the revocation changes are merged

Reply:
changes are merged https://github.com/wso2/carbon-apimgt/pull/7423

Reply:
added the revocation logic b861ac70300525979d3f146942ee522296f91a66